client/shaders/nodes_shader/opengl_fragment.glsl

```diff
@@ -117,7 +127,7 @@ float mtsmoothstep(in float edge0, in float edge1, in float x)
 
 float shadowCutoff(float x) {
 	#if defined(ENABLE_TRANSLUCENT_FOLIAGE) && MATERIAL_TYPE == TILE_MATERIAL_WAVING_LEAVES
-		return mtsmoothstep(0.0, 0.002, x);
+		return mtsmoothstep(0.0, 3.0 / f_shadowfar, x);
```

Comment:
Can you explain this one?

Reply:
Yes, the original code had an unintended problem where changing the shadow map distance made foliage not shaded correctly. This fixes that problem so it should look the same no matter the dynamic shadows settings.